packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2894,110 +2945,120 @@ export class ContainerRuntime
+	 * the message to the appropriate handler in the runtime.
 	 * Throws a DataProcessingError if the message looks like but doesn't conform to a known TypedContainerRuntimeMessage type.
+	 * @param messagesWithMetadata - A bunch of contiguous runtime messages and their metadata.
+	 * @param local - true if the messages were originally generated by the client receiving it.
+	 *
 	 */
-
-	private validateAndProcessRuntimeMessage(messageWithContext: {
-		message: InboundSequencedContainerRuntimeMessage;
-		local: boolean;
-		savedOp?: boolean;
-		localOpMetadata?: unknown;
-	}): void {
-		const { local, message, savedOp, localOpMetadata } = messageWithContext;
-
-		// Set the minimum sequence number to the containerRuntime's understanding of minimum sequence number.
-		if (
-			this.useDeltaManagerOpsProxy &&
-			this.deltaManager.minimumSequenceNumber < message.minimumSequenceNumber
-		) {
-			message.minimumSequenceNumber = this.deltaManager.minimumSequenceNumber;
-		}
-
-		this._processedClientSequenceNumber = message.clientSequenceNumber;
-
+	private validateAndProcessRuntimeMessages(
+		messagesWithMetadata: {
+			message: InboundSequencedContainerRuntimeMessage;
+			localOpMetadata: unknown;
+		}[],
+		local: boolean,
+		savedOp?: boolean,
+	): void {
 		// If there are no more pending messages after processing a local message,
 		// the document is no longer dirty.
 		if (!this.hasPendingMessages()) {
 			this.updateDocumentDirtyState(false);
 		}
 
-		switch (message.type) {
-			case ContainerMessageType.Attach:
-			case ContainerMessageType.Alias:
-			case ContainerMessageType.FluidDataStoreOp:
-				this.channelCollection.process(message, local, localOpMetadata);
-				break;
-			case ContainerMessageType.BlobAttach:
-				this.blobManager.processBlobAttachOp(message, local);
-				break;
-			case ContainerMessageType.IdAllocation:
-				// Don't re-finalize the range if we're processing a "savedOp" in
-				// stashed ops flow. The compressor is stashed with these ops already processed.
-				// That said, in idCompressorMode === "delayed", we might not serialize ID compressor, and
-				// thus we need to process all the ops.
-				if (!(this.skipSavedCompressorOps && savedOp === true)) {
-					const range = message.contents;
-					// Some other client turned on the id compressor. If we have not turned it on,
-					// put it in a pending queue and delay finalization.
-					if (this._idCompressor === undefined) {
-						assert(
-							this.idCompressorMode !== undefined,
-							0x93c /* id compressor should be enabled */,
-						);
-						this.pendingIdCompressorOps.push(range);
-					} else {
-						assert(
-							this.pendingIdCompressorOps.length === 0,
-							0x979 /* there should be no pending ops! */,
-						);
-						this._idCompressor.finalizeCreationRange(range);
+		const messagesType = messagesWithMetadata[0]?.message.type;
```

Comment:
The contract (that all messages are of same type) is not obvious and too easy to break.
It's not very clear why caller of this API (who knows that it has all the ops of the same type) does not call this.channelCollection.processChannelMessages directly, as opposed to delegating to this function.
Is this only due to optimization of having two call sites in processInboundMessages()?
One way to solve it - instead of using array to accumulate ops, we can use an object with flush() or push() method that  does it in one place.

Reply:
I do see an assert in switch statement below, so maybe it's not a big deal, but the contract feels weird.

Reply:
In the last iteration, there is only one message and its type. The contents are in an array so it looks much better now.